File: src/decoders/period.ts
import type { Period } from "~/models";
import { decodePronoteDate } from "./pronote-date";

export const decodePeriod = (period: any): Period => {
  return {
    id: period.N,
    kind: period.G,
    name: period.L,

    startDate: decodePronoteDate(period.dateDebut.V),
    endDate: decodePronoteDate(period.dateFin.V)
  };
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/period.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.Period

import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive

fun decodePeriod (period: JsonObject): Period {
    return Period(
        id = period["N"]!!.jsonPrimitive.content,
        kind = period["G"]!!.jsonPrimitive.content.toInt(),
        name = period["L"]!!.jsonPrimitive.content,

        startDate = decodePronoteDate(period["dateDebut"]!!.jsonObject["V"]!!.jsonPrimitive.content),
        endDate = decodePronoteDate(period["dateFin"]!!.jsonObject["V"]!!.jsonPrimitive.content)
    )
}